src/main.cpp

```diff
@@ -24,28 +27,45 @@ int main(int argc, char* argv[])
     for (auto i = 1; i < argc; i++) {
         const std::string param(argv[i]);
         if ((param == "--help") || (param == "-h")) {
-            openturbine::io::print_banner(std::cout);
-            openturbine::io::print_usage(std::cout);
+            io::print_banner(std::cout);
+            io::print_usage(std::cout);
             return 0;
         }
     }
 
-    int axis_size = 5;        // Size of the 1D grid
-    double side_length = 1.0; // Length of the 1D domain
-    double k = 1.011; // Thermal diffusivity of the material in units of cm^2/s
-    double ic_x0 = 100.0; // Initial temperature at the beginning of the domain
-    double ic_x1 = 100.0; // Initial temperature at the end of the domain
-    int n_max = 500;      // Max iterations
-    double residual_tolerance = 1e-5; // Iterative residual tolerance
+    // TODO Name the logging file based on the provided input file name
+    std::string log_file = "log.txt";
+    if (std::filesystem::exists(log_file)) {
+        std::cout << "Overwriting the previously existing log file" << std::endl;
+    }
+    std::ofstream{log_file};
+
+#ifdef DEBUG
```

Comment:
I wonder if this wouldn't be better to set as an input parameter. Whats the benefit to setting the severity level as a compiler flag? If it's an input parameter, you can enable/disable output for a given executable without having to go back and recompile.

Reply:
My opinion regarding the above is we are taking care of setting the log severity level based on the build type provided to `CMake`. If someone's using the `Release` version, they probably don't care about the `Debug` messages being printed (since there should be far more `debug` messages than `info` messages which are valuable to only devs). Using the debug mode as input to the logging utility unintentionally could come with a big performance hit. Adding the compiler flags based on the build type takes care of this issue rather than adding one more (redundant imo) input to the executable and opening up avenues for error. Feel free to tell me if it doesn't make sense.

Reply:
@rafmudaf I gave this a little more thought - I am open to the idea of setting the log level based on a parameter in the input file so it can be customized by the user. However, I think it will be useful mostly for `openturbine` developers and there exists such a switch for it in `CMakeLists.txt`. My opinion is to keep the current design and change things when there is such a feature request from users in the future.